head_first_design_patterns/tbruinem/json/src/include/parser.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+#include "common/string.hpp"
+#include "element_type.hpp"
+
+namespace json {
+
+class ParsedJSON;
+
+enum class ParseState {
+	OUT,	//! Initial state
+	WORD,	//! Name of a JSON item
+	VALUE,	//! Value of a JSON item
+	IN,		//! Between Name and Value
+};
```

Comment:
So when you say WORD/Name it's a state specifically for JsonObject keys and `IN` is just for the `:` character?
I would just name it as it is in that case :: `{INITIAL, OBJECT_KEY, JSON_VALUE, COLON}`